Description: Fix path in generate_payreqpb2.sh
From: Nicholas Voyak <nnnnnnnnnnnn@airmail.cc>
Forwarded: not-needed

--- a/contrib/generate_payreqpb2.sh
+++ b/contrib/generate_payreqpb2.sh
@@ -2,7 +2,7 @@
 # Generates the file paymentrequest_pb2.py
 
 CONTRIB="$(dirname "$(readlink -e "$0")")"
-EL="$CONTRIB"/../electrum
+EL="$CONTRIB"/../electrum_nmc/electrum
 
 if ! which protoc > /dev/null 2>&1; then
     echo "Please install 'protoc'"
