Description: Fix aiorpcx field
From: Nicholas Voyak <nnnnnnnnnnnn@airmail.cc>
Forwarded: not-needed

--- a/electrum_nmc/electrum/util.py
+++ b/electrum_nmc/electrum/util.py
@@ -1075,7 +1075,7 @@
 
     def spawn(self, *args, **kwargs):
         # don't complain if group is already closed.
-        if self._closed:
+        if self.joined:
             raise asyncio.CancelledError()
         return super().spawn(*args, **kwargs)

